chain/network/src/routing/routing_table_view.rs

```diff
@@ -99,11 +94,15 @@ impl RoutingTableView {
         }
     }
 
-    pub(crate) fn find_route(&mut self, target: &PeerIdOrHash) -> Result<PeerId, FindRouteError> {
+    pub(crate) fn find_route(
+        &mut self,
+        clock: &time::Clock,
```

Comment:
Note that this creates opens a new space for introducing bugs: if you pass around instant rather than clock, it is possible to store some instants and pass them instead of clock.now(). If you pass clock, such bugs are local - you have to store a timestamp close to the place where you will use it and you can shoot in the foot only yourself.

Reply:
It is especially important when you have some function which would panic on non-monotone clock reads.

Reply:
No strong opinion here! My biggest argument for passing instant is that it gives much better information to the reader: with instant, it's obvious that that's just some pure logic which cares about a timestamp. With clock, I begin to suspect that this is some concurrent thing, which will lock at the clock at different points in time. 

But I also agree with you point that, if you pass `now`, it might be too easy to pass the wrong now.

Reply:
ack